src/blocking.rs

```diff
@@ -0,0 +1,646 @@
+//! A blocking version of the SDK.
```

Comment:
I have mixed feelings about all blocking functionality living in this file. Did you consider breaking it up into separate, smaller files?

Reply:
I was planning to but @coltfred preferred them all together. I'm fine with doing it either way.